prqlc/prql-compiler/src/sql/gen_expr.rs

```diff
@@ -117,6 +117,7 @@ pub(super) fn translate_expr(expr: Expr, ctx: &mut Context) -> Result<ExprOrSour
             }
             super::operators::translate_operator_expr(expr, ctx)?
         }
+        ExprKind::Array(_) => panic!("unsupported expr kind: array"),
```

Comment:
Let's produce a proper error here.

Reply:
Done in [752a127](https://github.com/PRQL/prql/pull/3884/commits/752a1279ee01d8198f661516608d1709387672a4). Feedback welcome :)
Thank you for your review

Reply:
Looks great, thanks!